cobigen-plugins/cobigen-templateengines/cobigen-tempeng-freemarker/src/main/java/com/devonfw/cobigen/tempeng/freemarker/FreeMarkerTemplateEngine.java

```diff
@@ -73,8 +73,21 @@ public void process(TextTemplate template, Map<String, Object> model, Writer out
         env.setLogTemplateExceptions(false); // no duplicate logging
         env.process();
       } catch (TemplateException e) {
+        // Get root cause
+        Throwable rootCause = e;
+        while (rootCause.getCause() != null && rootCause.getCause() != rootCause) {
+          rootCause = rootCause.getCause();
+        }
```

Comment:
exception causes are recursively defined, meaning, you will not be able to get the root cause here. There is an Exception Util, which could be used to properly give you some better information.